src/NuGet.Core/NuGet.Protocol.Core.Types/NuGetHttpHeaderUtility.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Net.Http;
+using NuGet.Common;
+
+namespace NuGet.Protocol.Core.Types
+{
+    public class NuGetHttpHeaderUtility
+    {
+        public const string ApiKeyHeader = "X-NuGet-ApiKey";
+        public const string WarningHeader = "NuGet-Warning";
+
+        public static void LogServerWarning(ILogger log, HttpResponseMessage response)
```

Comment:
How about making it `HttpResponseMessageExtensions`?
Then calling this method would be more concise and intuitive
```c#
response.LogServerWarning(log)
```